### PR TITLE
DROOLS-1596: fixing checks of allowed values for duration and date and time values

### PR DESCRIPTION
**Pull request by etirelli**

### CODE REVIEW

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/InNode.java [hunk 1]
```diff
@@ -81,11 +81,12 @@ private Boolean in(EvaluationContext ctx, Object value, Object expr) {
         } else if ( expr instanceof UnaryTest ) {
             return ((UnaryTest) expr).apply( ctx, value );
         } else if ( expr instanceof Range ) {
-            if( !( value instanceof Comparable ) ) {
-                ctx.notifyEvt( astEvent(Severity.ERROR, Msg.createMessage(Msg.EXPRESSION_IS_RANGE_BUT_VALUE_IS_NOT_COMPARABLE)));
-                return null;
+            try {
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/InNode.java:84]: The reason for this change is that not every valid value implements Comparable. For instance, the java class Period does not implement Comparable, but it is a valid argument for ranges.

**Comment by etirelli** [comment]
@tarilabs can you please review?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/RangeNode.java [hunk 2]
```diff
@@ -93,9 +97,55 @@ public Range evaluate(EvaluationContext ctx) {
...
+            start = null;
+        }
+        return start;
+    }
+
+    public static class ComparablePeriod implements Comparable<Period> {
+        private final int left;
+
+        public ComparablePeriod(Period value) {
+            this.left = value.getYears() * 12 + value.getMonths();
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/RangeNode.java:127]: ok, normalizing to months

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/RangeNode.java [hunk 3]
```diff
@@ -93,9 +97,55 @@ public Range evaluate(EvaluationContext ctx) {
...
+        }
+
+        @Override
+        public int compareTo(Period o) {
+            int right = o.getYears() * 12 + o.getMonths();
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/RangeNode.java:132]: ok, normalizing to months

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/DecisionTableFunction.java [hunk 4]
```diff
@@ -161,11 +162,20 @@ private static DTDecisionRule toDecisionRule(EvaluationContext mainCtx, FEEL emb
         return dr;
     }
 
-    private static UnaryTest toUnaryTest(Object o) {
+    private static UnaryTest toUnaryTest(EvaluationContext ctx, Object o) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/DecisionTableFunction.java:165]: not urgent and not this PR, but something sounds wrong; I realising in many part we pass this ctx only for error reporting, and this is preventing us to use simple method references

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/InNode.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/ast/InNode.java:84]: ok

**Review by tarilabs — APPROVED** [review]

**Pull request merged**